common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -415,6 +421,109 @@ public void onReceive(@NonNull PropertyBag propertyBag) {
+        brokerActivityIntent.putExtra(BrokerActivity.BROKER_INTENT, interactiveATv2RequestIntent);
+
+        // Callback alias can stay the same in ATv2 case.
+        LocalBroadcaster.INSTANCE.registerCallback(RETURN_BROKER_INTERACTIVE_ACQUIRE_TOKEN_RESULT,
+                new LocalBroadcaster.IReceiverCallback() {
+                    @Override
+                    public void onReceive(@NonNull PropertyBag propertyBag) {
+                        /**
+                         * Get the response from the Broker captured by BrokerActivity.
+                         * BrokerActivity will pass along the response to the broker controller.
+                         * The Broker controller will map the response into the broker result
+                         * and signal the future with the broker result to unblock the request.
+                         */
+
+                        Logger.verbose(
+                                methodTag,
+                                "Received result (for ATv2 request) from Broker..."
+                        );
+
+                        Telemetry.emit(
+                                new ApiStartEvent()
+                                        .putApiId(TelemetryEventStrings.Api.BROKER_COMPLETE_ACQUIRE_TOKEN_ATV2_INTERACTIVE)
+                                        .put(TelemetryEventStrings.Key.REQUEST_CODE, propertyBag.<Integer>getOrDefault(REQUEST_CODE, -1).toString())
+                                        .put(TelemetryEventStrings.Key.RESULT_CODE, propertyBag.<Integer>getOrDefault(RESULT_CODE, -1).toString())
+                        );
+
+                        mBrokerResultFuture.setResult(PropertyBagUtil.toBundle(propertyBag));
+
+                        Telemetry.emit(
+                                new ApiEndEvent()
+                                        .putApiId(TelemetryEventStrings.Api.BROKER_COMPLETE_ACQUIRE_TOKEN_ATV2_INTERACTIVE)
+                        );
+
+                        LocalBroadcaster.INSTANCE.unregisterCallback(RETURN_BROKER_INTERACTIVE_ACQUIRE_TOKEN_RESULT);
+                    }
+                });
+
+        // To support calling from OneAuth-MSAL, which may be initialized without an Activity
+        // add Flags to start as a NEW_TASK if we are launching from an application Context
+        brokerActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        mApplicationContext.startActivity(brokerActivityIntent);
+
+        final AcquireTokenResult result;
+        try {
+            // Wait to be notified of the result being returned... we could add a timeout here if we want to
+            final Bundle resultBundle = mBrokerResultFuture.get();
+
+            final String negotiatedBrokerProtocolVersion = interactiveATv2RequestIntent.getStringExtra(NEGOTIATED_BP_VERSION_KEY);
+
+            // TODO: Is this accurate? BrokerAccountDataManager contains MSA accounts
```

Comment:
Ping

Reply:
This should not be needed anymore. This was before PRTv3 when MSA accounts where not supported in broker.